xrpl/caip122.md

```diff
@@ -0,0 +1,112 @@
+Correspondingly, a message needs to be serialized to bytes prior to signing it with the [Ed25519][] algorithm.
+
+### Signature Type
+
+We propose using the signature type `xrpl:secp256k1` and `xrpl:ed25519` to refer to the chain and algorithm used uniquely.
+
+### Signature Creation
+
+The abstract data model must be canonicalized to a string representation in an unambigious format, and then the string serialized to the proper binary format mentioned [above](#signing-algorithm) to be signed over with the proper algorithm. 
+Depending on the method or libraries used for signatures, this conversion may or may not happen automatically. 
+For example, if using [ripple-keypairs][], you will need to convert the key to a byte representation before signing using [Ed25519][] and for [secp256k1][] the library will handle hashing and signing internally.
+
+We propose the following string format, inspired by the ABNF in [EIP-4361][].
+
+```
+${domain} wants you to sign in with your XRPL account:
+${address}
+
+${statement}
+
+URI: ${uri}
+Version: ${version}
+Chain ID: ${chain-id}
+Nonce: ${nonce}
+Issued At: ${issued-at}
+Expiration Time: ${expiration-time}
+Not Before: ${not-before}
+Request ID: ${request-id}
+Resources:
+- ${resources[0]}
+- ${resources[1]}
+...
+- ${resources[n]}
+```
+
+For example,
+
+```
+service.org wants you to sign in with your XRPL account:
+r4FTvnahbUfhe1WK2EK5Jz4cNvdFvT8Dzt
+
+I accept the ServiceOrg Terms of Service: https://service.org/tos
+
+URI: https://service.org/login
+Version: 1
+Chain ID: 0
+Nonce: 32891757
+Issued At: 2021-09-30T16:25:24.000Z
+Resources:
+- ipfs://Qme7ss3ARVgxv6rXqVPiikMJ8u2NLgmgszg13pYrDKEoiu
+- https://example.com/my-web2-claim.json
+```
+
+The preceeding example would be hashed to `4beb3f7d5c8bf8deab467de033628274a9053f688f82e0f023cddcb0564cfc9a` prior to being signed using the [secp256k1][] algorithm.
+
+The serialization of this string prior to signing with [Ed25519][Ed25519] would produce:
+```
+736572766963652e6f72672077616e747320796f7520746f207369676e20696e207769746820796f7572205852504c206163636f756e743a0a72344654766e6168625566686531574b32454b354a7a34634e766446765438447a740a0a49206163636570742074686520536572766963654f7267205465726d73206f6620536572766963653a2068747470733a2f2f736572766963652e6f72672f746f730a0a5552493a2068747470733a2f2f736572766963652e6f72672f6c6f67696e0a56657273696f6e3a20310a436861696e2049443a20300a4e6f6e63653a2033323839313735370a4973737565642041743a20323032312d30392d33305431363a32353a32342e3030305a0a5265736f75726365733a0a2d20697066733a2f2f516d653773733341525667787636725871565069696b4d4a3875324e4c676d67737a673133705972444b456f69750a2d2068747470733a2f2f6578616d706c652e636f6d2f6d792d776562322d636c61696d2e6a736f6e
+```
+### Signature Verification
+
+Signature verification behaves similarly. We can use standard [secp256k1][] or[Ed25519][] signature verification. 
```

Comment:
Note that in the case of EVM, there are multiple signing methods (the spec mentioned eip155, aka `personal_sign`, classic metamask arbitrary unstructured text signing, and eip1271, aka `smart contract wallet signing`, for signatures produced by deployed smart contracts).  is there only one signing method in Ripple, that you simply pass a secp256k1 or ed25519 key to, or are they two different signing interfaces with slightly different digest constraints/encodings, etc? 

if the latter, it would be good to define the ENUM of valid `signatureMeta.t` values (recognizable names of signature interfaces in the XRP toolchain). and if the former, maybe just an ENUM with only one member? 😄

Reply:
https://xrpl.org/signing-methods.html

I hadn't thought about multi-signatures, but maybe we just include that in the enum and let future teams worry about that one 😅

Reply:
There is only one signing method in Ripple, that you simply pass a secp256k1 or ed25519 key.

Reply:
i mean, multi-signature might be the same algorithm/method but isn't the payload a little differently constructed? or am i missing something? you don't need to define it, I just want to make sure someone else defining it later understands how to do it (i.e., how to PR this adding a section for what's particular to it and adding a new valid value for `SignatureMeta.t`)

Reply:
I don't think multi-signature is relevant for this use case. But I'm happy to learn more about it otherwise.